cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -50,32 +50,16 @@ def _validate_device_specification(proto: v2.device_pb2.DeviceSpecification) ->
                         " which is not in valid_qubits."
                     )
 
-        # Symmetric and asymmetric targets should not have repeated qubits.
-        if (
-            target_set.target_ordering == v2.device_pb2.TargetSet.SYMMETRIC
-            or target_set.target_ordering == v2.device_pb2.TargetSet.ASYMMETRIC
-        ):
+        # Symmetric targets should not have repeated qubits.
+        if target_set.target_ordering == v2.device_pb2.TargetSet.SYMMETRIC:
```

Comment:
Should we raise an error if target set is not symmetric?

Reply:
The more general question is whether we should raise an error if deprecated fields are set. Because some deprecated fields are still set for backward compatibility, raising an error I think would be disruptive to the user experience. In this case ASYMMETRIC isn't used, but it's still technically allowed, so I'm leaning towards following the same principle.

Reply:
Yes, it can be set, but it is not a valid value for a grid device.  If you set a a TargetSet as ASYMMETRIC, the grid device will not respect it.  Silently not doing what the user wants is not a good practice, as it hides the fact that the grid device is not following the spec the user provided.  I think this should throw an exception since it is not a supported value for the grid device.

Reply:
Discussed offline - will include an exception.